packages/plugin-team/src/team.js

```diff
@@ -0,0 +1,369 @@
+      })
+      .then((a) => this.spark.conversation.submit(a));
+  },
+
+  /**
+   * Update the displayName, summary, or teamColor field for a team.
+   * @param {TeamObject} team to be updated
+   * @param {Object} object with updated displayName, summary, and/or teamColor
+   * @param {Object} activity Reference to the activity that will eventually be
+   * posted. Use this to (a) pass in e.g. clientTempId and (b) render a
+   * provisional activity
+   * @returns {Promise} Resolves with posted activity
+   */
+  update(team, object, activity) {
+    return this.spark.conversation.update(team, object, activity);
+  },
+
+  _ensureGeneralConversation(team) {
+    if (!team.generalConversationUuid) {
+      return Promise.reject(new Error(`\`team.generalConversationUuid\` must be present`));
+    }
+
+    if (team.conversations.items.length) {
+      const generalConversation = find(team.conversations.items, {id: team.generalConversationUuid});
+
+      if (generalConversation) {
+        return Promise.resolve(generalConversation);
+      }
+    }
+
+    return this.spark.conversation.get({id: team.generalConversationUuid});
+  },
+
+  _inferTeamUrl(team) {
+    if (!team.url && team.id) {
+      return this.spark.device.getServiceUrl(`conversation`)
+        .then((url) => {
+          team.url = `${url}/teams/${team.id}`;
+          /* istanbul ignore else */
+          if (process.env.NODE_ENV !== `production`) {
+            this.logger.warn(`team: inferred team url from conversation id; please pass whole team objects to Team methods`);
+          }
+          return team;
+        });
+    }
+
+    return Promise.resolve(team);
+  },
+
+  _prepareTeam: function _prepareTeam(params) {
+    const payload = this.spark.conversation._prepareConversationForCreation(params);
+
+    payload.objectType = `team`;
+
+    if (params.summary) {
+      payload.summary = params.summary;
+    }
+
+    return Promise.resolve(payload);
+  },
+
+  _prepareTeamConversation(teamConversation, params) {
+    if (!teamConversation.kmsResourceObjectUrl) {
+      return Promise.reject(new Error(`Team general conversation must have a KRO`));
+    }
+
+    const payload = this.spark.conversation._prepareConversationForCreation(params);
+
+    // Push the general conversation KRO on so that team members can
+    // decrypt the title of this open room without joining it.
+    payload.kmsMessage.userIds.push(teamConversation.kmsResourceObjectUrl);
+    return Promise.resolve(payload);
+  },
+
+  _recordUUIDs(team) {
+    if (!team.teamMembers || !team.teamMembers.items) {
+      return Promise.resolve(team);
+    }
+
+    return Promise.all(team.teamMembers.items.map((participant) => this.spark.user.recordUUID(participant)));
+  }
+
+});
+
+/**
+ * Assign or unassign a team member to be a moderator of a team
+ * @param {TeamObject} team
+ * @param {Object} participant
+ * @param {Object} activity
+ * @returns {Promise} Resolves with activity that was posted
+ */
+[
```

Comment:
It will create two method under Team.
Team.assignModerator
Team.unassignModerator
Each method will call submitModerationChangeActivity, which in turn calls ...spark.conversation.assignModerator or ...spark.conversation.unassignModerator respectively